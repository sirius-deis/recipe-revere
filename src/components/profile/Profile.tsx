import { FaUserSecret } from "react-icons/fa";
import Col from "../column/Col";
import Row from "../row/Row";
import styles from "./Profile.module.css";
import { FC } from "react";
import { Link } from "react-router-dom";
import SavedRecipeList from "../savedRecipeList/SavedRecipeList";
import FriendsActivityList from "../friendsActivityList/FriendsActivityList";

interface ProfileProps {
  user: {
    _id: string;
    name: string | undefined;
    isActive: boolean;
    isBlocked: boolean;
    pictures: string[];
  };
}

const Profile: FC<ProfileProps> = ({ user }) => {
  return (
    <div className={styles.profile}>
      <h1 className="text-left">My Profile</h1>
      <div className={styles["profile-header"]}>
        <Col>
          <Row inlineStyles={{ gap: "1rem" }}>
            <div className={styles["image-container"]}>
              {user?.pictures.length === 0 ? (
                <FaUserSecret className={styles["user-icon"]} />
              ) : (
                <img data-testid="img" src={`${user?.pictures![0]}`} alt="user profile" />
              )}
            </div>
          </Row>
          <Row>
            <h2>{user?.name || "Anonymous User"}</h2>
          </Row>
        </Col>
      </div>
      <section>
        <Row inlineStyles={{ justifyContent: "space-between" }}>
          <Col>
            <h2>Saved recently</h2>
          </Col>
          <Col>
            <Link to="/saved">See all</Link>
          </Col>
        </Row>
        <SavedRecipeList />
      </section>
      <section>
        <h2 className="text-left">Friends activity</h2>
        <FriendsActivityList />
      </section>
    </div>
  );
};

export default Profile;
