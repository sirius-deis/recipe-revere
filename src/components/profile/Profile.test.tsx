import { render, screen } from "@testing-library/react";
import { BrowserRouter as Router } from 'react-router-dom';
import Profile from "./Profile";

jest.mock("../savedRecipeList/SavedRecipeList.tsx", () => () => <div>SavedRecipeList</div>)
jest.mock("../friendsActivityList/FriendsActivityList.tsx", () => () => <div>FriendsActivityList</div>)

describe("Profile component", () => {
  const mockedUser = {
    _id: "123",
    name: "John Doe",
    isActive: true,
    isBlocked: false,
    pictures: ["https://example.com/profile.jpg"],
  };
  it("should match snapshot", () => {
    const { container } = render(<Router><Profile user={mockedUser} /></Router>);
    expect(container).toMatchSnapshot();
  })
  it("should render user profile with provided user data", () => {
    render(<Router><Profile user={mockedUser} /></Router>);
    expect(screen.getByText(mockedUser.name)).toBeInTheDocument()
    expect(screen.queryByText("Anonymous User")).not.toBeInTheDocument()
    expect(screen.getByTestId("img")).toHaveAttribute("src", mockedUser.pictures[0])
  })
  it("should render 'Anonymous User' when user name is not provided", () => {
    render(<Router><Profile user={{ ...mockedUser, name: undefined }} /></Router>);
    expect(screen.getByText("Anonymous User")).toBeInTheDocument()
  })
})