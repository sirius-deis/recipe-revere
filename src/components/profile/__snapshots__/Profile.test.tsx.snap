// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component should match snapshot 1`] = `
<div>
  <div
    class="profile"
  >
    <h1
      class="text-left"
    >
      My Profile
    </h1>
    <div
      class="profile-header"
    >
      <div
        class="col"
      >
        <div
          class="row"
          style="gap: 1rem;"
        >
          <div
            class="image-container"
          >
            <img
              alt="user profile"
              data-testid="img"
              src="https://example.com/profile.jpg"
            />
          </div>
        </div>
        <div
          class="row"
        >
          <h2>
            John Doe
          </h2>
        </div>
      </div>
    </div>
    <section>
      <div
        class="row"
        style="justify-content: space-between;"
      >
        <div
          class="col"
        >
          <h2>
            Saved recently
          </h2>
        </div>
        <div
          class="col"
        >
          <a
            href="/saved"
          >
            See all
          </a>
        </div>
      </div>
      <div>
        SavedRecipeList
      </div>
    </section>
    <section>
      <h2
        class="text-left"
      >
        Friends activity
      </h2>
      <div>
        FriendsActivityList
      </div>
    </section>
  </div>
</div>
`;
